contracts/SnapshotToken/Extensions/SimpleVote.sol

```diff
@@ -0,0 +1,191 @@
+
+        _proposals[_nextProposalId++] = Proposal({
+            inFavor: 0,
+            against:0,
+            endTime: now + VOTING_PERIOD,
+            owner: msg.sender,
+            snapshotId: sId
+        });
+
+        _hasActiveProposal[msg.sender] = true;
+
+        emit LogNewProposal(msg.sender, _nextProposalId - 1, sId, now + VOTING_PERIOD);
+    }
+
+    /// @dev increase the votecount on a given proposal by the token balance of the sender
+    ///   throws if proposal does not exist or the vote on it has ended already. Votes are final,
+    ///   changing the vote is not allowed
+    /// @param proposalId of the proposal to be voted on
+    /// @param voteInFavor of the desired proposal
+    function vote(uint256 proposalId, bool voteInFavor) public {
+        incrementVote(proposalId, voteInFavor, msg.sender);
+    }
+
+    /// @dev same as vote, only for a relayed vote. Will throw if provided signature (v,r,s) does not match
+    ///  the address of the voter
+    /// @param voter address whose token balance should be used as voting weight
+    function relayedVote(
+        uint256 proposalId,
+        bool voteInFavor,
+        address voter,
+        bytes32 r,
+        bytes32 s,
+        uint8 v
+    )
+        public
+    {
+        // check that message signature matches the voter address
+        // solium-disable indentation
+        require(ecrecover(
+            keccak256(abi.encodePacked(
+                "\x19Ethereum Signed Message:\n32",
+                keccak256(abi.encodePacked(byte(0), this, proposalId, voteInFavor)))),
+            v, r, s) == voter,
+        "Incorrect order signature");
+        // solium-enable indentation
+
+        incrementVote(proposalId, voteInFavor, voter);
+    }
+
+    function batchRelayedVotes(
+        uint256[] proposalIndices,
+        bool[] votePreferences,
+        address[] voters,
+        bytes32[] r,
+        bytes32[] s,
+        uint8[] v
+    )
+        public
+    {
+        require(
+            proposalIndices.length == votePreferences.length &&
+            votePreferences.length == voters.length &&
+            voters.length == r.length &&
+            r.length == s.length &&
+            s.length == v.length,
+            "Invalid voting arguments"
+        );
+        for (uint i = 0; i < voters.length; i++) {
+            relayedVote(
+                proposalIndices[i],
+                votePreferences[i],
+                voters[i],
+                r[i],
+                s[i],
+                v[i]
+            );
+        }
+    }
+
+    /// @return whether the vote on a proposal has passed, if false this could be because it was rejected or
+    ///   because the quorum was not reached.
+    /// @dev throws if vote is still ongoing or proposal does not exist. Also deletes the proposalData to free up gas.
+    //     Therefore subsequent attempts to call this function will throw too
+    function getResult(uint256 proposalId) public {
```

Comment:
practical note:
a `constant` version of this function would be pretty useful to dapp developers (if we want to show results during the voting).

Reply:
yeah, here my design choice, was to not make intermediate results easily accessible and to save gas